app/assets/javascripts/discourse/app/models/pending-post.js

```diff
@@ -0,0 +1,42 @@
+import discourseComputed from "discourse-common/utils/decorators";
+import Category from "discourse/models/category";
+import RestModel from "discourse/models/rest";
+import { userPath } from "discourse/lib/url";
+import { alias } from "@ember/object/computed";
+import { cookAsync } from "discourse/lib/text";
+import { loadOneboxes } from "discourse/lib/load-oneboxes";
+import { ajax } from "discourse/lib/ajax";
+import { resolveAllShortUrls } from "pretty-text/upload-short-url";
+
+const PendingPost = RestModel.extend({
+  expandedExcerpt: null,
+  postUrl: alias("topic_url"),
+
+  init() {
+    this._super(...arguments);
+    cookAsync(this.raw_text).then((cooked) => {
```

Comment:
Random thought: if we display 10 pending posts, does this mean we'll be doing 10 `loadOneboxes` and 10 `resolveAllShortUrls` to the server? Are they already cached on the client?